py-polars/polars/lazyframe/frame.py

```diff
@@ -932,6 +937,201 @@ def pipe(
         """
         return function(self, *args, **kwargs)
 
+    def describe(
+        self,
+        percentiles: Sequence[float] | float | None = (0.25, 0.50, 0.75),
+        *,
+        interpolation: RollingInterpolationMethod = "nearest",
+    ) -> DataFrame:
+        """
+        Creates a summary of statistics for a LazyFrame, returning a DataFrame.
```

Comment:
There should probably be a big warning that says that using this function will execute your query and potentially be expensive.

I would recommend including a `Returns` section for this specific method, just so it's extremely clear it returns a DataFrame.

Alternatively, this could return a LazyFrame and require a collect afterwards.

Reply:
Good idea; I moved the note about it losing laziness into the `warnings` section, and made it a bit more explicit 👌 (and added `Returns`)